### PR TITLE
Set close-on-exec on by default

### PR DESCRIPTION
**Pull request by mathias-lang-sociomantic**
This was introduced in #67 

CC @david-eckardt-sociomantic 
CC @mihails-strasuns-sociomantic for v4.0.0

### CODE REVIEW

**Review by nemanja-boric-sociomantic — APPROVED** [review]

File: src/ocean/sys/CloseOnExec.d [hunk 1]
```diff
@@ -53,11 +53,9 @@
     This does not affect `stdin`, `stdout` and `stderr`, which are opened before
     the start of the program.
 
-    Currently the flag is `false` to keep the system's default behaviour. In
-    general it is recommended to set it to `true` to avoid the aforementioned
-    problem unless file descriptor inheriting is needed. Since there appears to
-    be no use case for it for us the default value of `open_with_close_on_exec`
-    will change to `true` in the next major ocean release.
+    Currently the flag is `true`, which is in general recommended to avoid the
```
Comment (david-eckardt-sociomantic) [inline src/ocean/sys/CloseOnExec.d]: Please remove the word "Currently", I wrote it to imply that this will change in a future release.

**Comment by mathias-lang-sociomantic** [comment]
```diff
diff --git a/src/ocean/sys/CloseOnExec.d b/src/ocean/sys/CloseOnExec.d
index 26246a3d..7dfbb325 100644
--- a/src/ocean/sys/CloseOnExec.d
+++ b/src/ocean/sys/CloseOnExec.d
@@ -53,7 +53,7 @@
     This does not affect `stdin`, `stdout` and `stderr`, which are opened before
     the start of the program.
 
-    Currently the flag is `true`, which is in general recommended to avoid the
+    The flag is `true` by default, which is in general recommended to avoid the
     aforementioned problem unless file descriptor inheriting is needed, which
     is not a feature that is currently relevant to Sociomantic's use cases.
 
```

**Review by david-eckardt-sociomantic — APPROVED** [review]

**Review by mihails-strasuns-sociomantic — APPROVED** [review]
Rebased on current v4.x.x head, waiting for CI

File: relnotes/cloonexec.migration.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+* `ocean.sys.CloseOnExec`
```
Comment (leandro-lucarella-sociomantic) [inline relnotes/cloonexec.migration.md:1]: This format is wrong. It should be:
```md
## Title

`module`

Description
```

**Review by leandro-lucarella-sociomantic — COMMENTED** [review]
@mathias-lang-sociomantic can you submit a PR fixing all the changelog formatting issues? :innocent: 

I very recently submitted a PR fixing all the changelog and now is all broken again :cry:

**Comment by mathias-lang-sociomantic** [comment]
> @mathias-lang-sociomantic can you submit a PR fixing all the changelog formatting issues? :innocent:

Sure, just make sure you stop changing the format :P

**Comment by leandro-lucarella-sociomantic** [comment]
It was changed only once!

**Pull request merged**